src/rate/index.tsx

```diff
@@ -0,0 +1,343 @@
+import { DNode, RenderResult } from '@dojo/framework/core/interfaces';
+import { uuid } from '@dojo/framework/core/util';
+import { create, tsx } from '@dojo/framework/core/vdom';
+import focus from '@dojo/framework/core/middleware/focus';
+import i18n from '@dojo/framework/core/middleware/i18n';
+import { createICacheMiddleware } from '@dojo/framework/core/middleware/icache';
+import Icon from '../icon';
+import Radio from '../radio';
+import RadioGroup from '../radio-group';
+import theme from '../middleware/theme';
+import * as baseCss from '../common/styles/base.m.css';
+import * as iconCss from '../theme/default/icon.m.css';
+import * as radioCss from '../theme/default/radio.m.css';
+import * as radioGroupCss from '../theme/default/radio-group.m.css';
+import * as css from '../theme/default/rate.m.css';
+import bundle from './nls/Rate';
+
+export interface MixedNumber {
+	quotient: number;
+	numerator: number;
+	denominator: number;
+}
+
+export interface RateProperties {
+	/** The name attribute for this rating group */
+	name: string;
+	/** The label to be displayed in the legend */
+	label?: string;
+	/** The id used for the radio input elements */
+	widgetId?: string;
+	/** The initial rating value */
+	initialValue?: number;
+	/** Callback fired when the rating value changes */
+	onValue?: (value?: number, mixedValue?: MixedNumber) => void;
+	steps?: number;
+	max?: number;
+	allowClear?: boolean;
+	disabled?: boolean;
+	readOnly?: boolean;
+}
+
+export interface RateChildren {
+	(fill: boolean, integer: number, selected?: number, over?: number): RenderResult;
+}
+
+interface RateState {
+	uuid: string;
+	hover: string;
+	hovering: boolean;
+	selectedInteger?: number;
+	selectedStep?: number;
+	focused: boolean;
+}
+
+const icache = createICacheMiddleware<RateState>();
+
+function mixedNumber(integer: number, step: number, steps: number) {
+	if (step === steps) {
+		step = 0;
+	} else {
+		integer -= 1;
+	}
+	return { quotient: integer, numerator: step, denominator: steps };
+}
+
+const factory = create({ focus, i18n, icache, theme })
+	.properties<RateProperties>()
+	.children<RateChildren | undefined>();
+
+export const Rate = factory(function Rate({
+	children,
+	properties,
+	middleware: { focus, i18n, icache, theme }
+}) {
+	const { format } = i18n.localize(bundle);
+	const {
```

Comment:
could you just use `const themedCss = theme.classes(css)` as this is what we do elsewhere and keeps things clean and consistent. It's hard to reason with the source of the classes when they have been deconstructed like this.